docs/contribute/test/testing.md

```diff
@@ -0,0 +1,139 @@
+  return RUN_ALL_TESTS();
+}
+```
+
+## Where to add the tests?
+
+After writing tests for your specific modules or plugins, you can add them under `https://github.com/WasmEdge/WasmEdge/tree/master/test` and for more specific placement of the tests, 
+
+For example,
+
+- You can add the tests for new plugins or update existing ones under `https://github.com/WasmEdge/WasmEdge/tree/master/test/plugins`
+
+## Documenting the Tests in Markdown
+
+When documenting these tests in Markdown, you should focus on explaining the purpose of each test, the setup required, and the expected outcomes. Here's an example of how you might structure this documentation:
+
+
+```md
+# WasmEdge Module Tests
+
+## Overview
+This document outlines the tests for the `WasiLogging` module in a WebAssembly environment using the WasmEdge runtime.
+
+## Test Setup
+- **Module Creation**: A module instance is created for testing purposes using the `createModule` function.
+- **Memory Manipulation**: The `fillMemContent` functions are used to manipulate memory content, essential for setting up test scenarios.
+
+## Test Cases
+
+### `func_log` Tests
+- **Purpose**: These tests validate the logging functionality at different levels and conditions.
+- **Test Scenarios**:
+  - **Clear Memory**: Ensures memory is correctly cleared before each test.
+  - **Set Strings in Memory**: Tests the ability to write strings to memory.
+  - **Logging at Different Levels**: Validates logging behavior at various levels (info, warning, error, etc.).
+  - **Stderr Context**: Checks if the logging correctly handles the stderr context.
+  - **Unknown Level**: Ensures that an unknown logging level is handled appropriately.
+
+## Running Tests
+- Compile the test code with your C++ compiler targeting WebAssembly.
+- Run the tests using a command-line tool that supports Google Test.
+
+## Conclusion
+These tests ensure the robustness and reliability of the module in a WebAssembly environment
+
+```
+
+### Further Reading
+
+- Here is an example test case explaination for [test/plugins/wasi_logging](./example.md)
+- Red Hat and WebAssembly: This blog post by Red Hat discusses WebAssembly (WASM) and its applications - [Link](https://www.redhat.com/en/blog/red-hat-and-webassembly)
+- Complete Docker + Wasm Tutorial: From C++ Code to Wasm - [Link](https://kodekloud.com/blog/docker-wasm-tutorial/)
+- How I created a SSR React App using WebAssembly (Wasm) - [Link](https://medium.com/wasm/how-i-created-a-ssr-react-app-using-webassembly-wasm-a6541431d0cd)
+
+#### Youtube Channels
+
+- [TheCherno](https://www.youtube.com/playlist?list=PLlrATfBNZ98dudnM48yfGUldqGD0S4FFb): This channel provides comprehensive tutorials on C++ programming, including aspects of testing and debugging.
+
+- [CppCon](https://www.youtube.com/user/CppCon): CppCon is a conference for C++ enthusiasts and professionals. Their YouTube channel features talks from experts, some of which cover testing methodologies in C++.
```

Comment:
The guide is not to let people learn C++

Reply:
CPPcon is a huge conference. Why we mention some materials  that user hardly to find

Reply:
Removed it